tiling.py

```diff
@@ -25,18 +28,43 @@ def __init__(self, height, up, mirror1_ctr, mirror2_ctr):
             shapes.Plane(ctr1, ctr1 + up_proj, ctr1 + right_proj),
             shapes.Plane(ctr2, ctr2 + up_proj, ctr2 + right_proj)
         ]
+
+        #add floor mirror plane if we only have floor mirror else add ceiling mirror plane
+        if not self._flag_ceiling and self._flag_floor:
+            self._vmirror_planes = shapes.Plane((0, 0, 1), (0, 0, 0), (1, 0, 0))
+        else:
+            self._vmirror_planes = shapes.Plane((0, height, 1), (0, height, 0), (1, height, 0))
+
         self._dihedral_transforms = [np.eye(4), utils.reflection_matrix(self._mirror_planes[0])]
-        self._translational_basis = [2*m1n*width, 2*m2n*width]
+
+        #there are 4 fds in tfd with ceiling or floor mirror
+        if self._flag_ceiling or self._flag_floor:
+            self._dihedral_transforms = []
+            last_transform = np.identity(4)
+            for i in range(4):
+                self._dihedral_transforms.append(last_transform)
+                p = copy.deepcopy([self._mirror_planes[0], self._vmirror_planes][i % 2])
+                p.transform(last_transform)
+                last_transform = np.dot(utils.reflection_matrix(p), last_transform)
+
+        self._translational_basis = [2*m1n*width, 2*m2n*width, 2*height*up_proj]
```

Comment:
This is incorrect.
The translational basis includes all vectors in which the tile could translate (e.g. x and -x). Also if only the floor or only the ceiling is present, there is no additional translation needed. 

```
self._translational_basis = [2*m1n*width, 2*m2n*width]

if self._floor and self._ceiling:
    self._translational_basis.extend([up*height, -up*height])
```